intellij/src/saros/intellij/filesystem/IntelliJReferencePointManager.java

```diff
@@ -0,0 +1,205 @@
+   * @exception IllegalArgumentException if the {@link Module} module or the {@link IReferencePoint}
+   *     reference point is null
+   */
+  public void putIfAbsent(@NotNull IReferencePoint referencePoint, @NotNull Module module) {
+    referencePointToModuleMapper.putIfAbsent(referencePoint, module);
+  }
+
+  /**
+   * Returns the {@link Module} module given by the {@link IReferencePoint} reference point
+   *
+   * @param referencePoint the key for which the module should be returned
+   * @return the module given by referencePoint
+   * @exception IllegalArgumentException if the {@link IReferencePoint} reference point is null
+   */
+  public @NotNull Module getModule(@NotNull IReferencePoint referencePoint) {
+    Module module = referencePointToModuleMapper.get(referencePoint);
+
+    if (module == null)
+      throw new IllegalArgumentException(
+          "For reference point " + referencePoint + " doesn't exist a module.");
+
+    return module;
+  }
+
+  /**
+   * Returns the {@link VirtualFile} resource in combination of the {@link IReferencePoint}
+   * reference point and the {@link IPath} relative path from the reference point to the resource.
+   *
+   * @param referencePoint The reference point, on which the resource belongs to
+   * @param referencePointRelativePath the relative path from the reference point to the resource
+   * @return the virtualFile of the reference point from referencePointRelativePath
+   * @exception IllegalArgumentException if for {@link IReferencePoint} reference point doesn't
+   *     exists a module or the {@link IPath} relative path is null
+   */
+  public VirtualFile getResource(
+      @NotNull IReferencePoint referencePoint, @NotNull IPath referencePointRelativePath) {
+    Module module = getModule(referencePoint);
+
+    return FilesystemUtils.findVirtualFile(module, referencePointRelativePath);
+  }
+
+  /**
+   * Returns the {@link IResource} resource in combination of the {@link IReferencePoint} reference
+   * point and the {@link IPath} relative path from the reference point to the resource.
+   *
+   * @param referencePoint The reference point, on which the resource belongs to
+   * @param referencePointRelativePath the relative path from the reference point to the resource
+   * @return the resource of the reference point from referencePointRelativePath
+   * @exception IllegalArgumentException if for {@link IReferencePoint} reference point doesn't
+   *     exists a module or the {@link IPath} relative path is null
+   */
+  public @NotNull IResource getSarosResource(
+      @NotNull IReferencePoint referencePoint, @NotNull IPath referencePointRelativePath) {
+    Module module = getModule(referencePoint);
+    VirtualFile vFile = getResource(referencePoint, referencePointRelativePath);
+
+    return VirtualFileConverter.convertToResource(module.getProject(), vFile);
+  }
+
+  private static class FilesystemUtils {
+
+    @Inject private static Project project;
```

Comment:
Ok, I understand. There is only one case, when I need the `Project`: If an `IReferencePoint` should be created, if you only have a `VirtualFile (`IReferencePoint create(
      @NotNull VirtualFile virtualFile`). I extended this to `
IReferencePoint create(
      @NotNull VirtualFile virtualFile, @NotNull SarosSession sarosSession)
`, so that the `IReferencePointManager` is able to get the `Project` with the help of the `SharedIDEContext`, which the `IReferencePointManager` gets from the `SarosSession`.